src/main/java/org/zalando/nakadi/domain/SubscriptionBase.java

```diff
@@ -126,4 +127,8 @@ public boolean equals(final Object o) {
     public int hashCode() {
         return Objects.hash(owningApplication, eventTypes, consumerGroup, readFrom, initialCursors);
     }
+
+    public Resource<SubscriptionBase> asBaseResource() {
+        return new ResourceImpl<>(null, ResourceImpl.SUBSCRIPTION_RESOURCE, getAuthorization(), this);
```

Comment:
If resource name is supposed to be null - then it should be defined as `@Nullable`. 
But it would be strange to make it nullable as Resource interface uses Optional in another place and usually it's not good to use several approaches to represent no-value. 

So my suggestion is:
- make resource name optional
or
- for this specific case provide some predefined resource name like "new-subscription"

Reply:
Resolved